receiver/prometheusexecreceiver/receiver_test.go

```diff
@@ -113,3 +123,688 @@ func assertTwoUniqueValuesScraped(t *testing.T, metricsSlice []pdata.Metrics) {
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range configTests {
+		t.Run(test.name, func(t *testing.T) {
+			got := getSubprocessConfig(test.config)
+			if !reflect.DeepEqual(got, test.wantSubprocessConfig) {
+				t.Errorf("getSubprocessConfig() got = %+v, want %+v", got, test.wantSubprocessConfig)
+			}
+		})
+	}
+}
+
+func TestExtractName(t *testing.T) {
+	customNameTests := []struct {
+		name   string
+		config *Config
+		want   string
+	}{
+		{
+			name: "no custom name",
+			config: &Config{
+				ReceiverSettings: configmodels.ReceiverSettings{
+					TypeVal: "prometheus_exec",
+					NameVal: "prometheus_exec",
+				},
+				ScrapeInterval: 60 * time.Second,
+				Port:           9104,
+				SubprocessConfig: subprocessmanager.SubprocessConfig{
+					Command: "mysqld_exporter",
+					Env:     []subprocessmanager.EnvConfig{},
+				},
+			},
+			want: "prometheus_exec",
+		},
+		{
+			name: "no custom name, only trailing slash",
+			config: &Config{
+				ReceiverSettings: configmodels.ReceiverSettings{
+					TypeVal: "prometheus_exec",
+					NameVal: "prometheus_exec/",
+				},
+				ScrapeInterval: 60 * time.Second,
+				Port:           9104,
+				SubprocessConfig: subprocessmanager.SubprocessConfig{
+					Command: "mysqld_exporter",
+					Env:     []subprocessmanager.EnvConfig{},
+				},
+			},
+			want: "prometheus_exec",
+		},
+		{
+			name: "custom name",
+			config: &Config{
+				ReceiverSettings: configmodels.ReceiverSettings{
+					TypeVal: "prometheus_exec",
+					NameVal: "prometheus_exec/custom",
+				},
+				ScrapeInterval: 60 * time.Second,
+				Port:           9104,
+				SubprocessConfig: subprocessmanager.SubprocessConfig{
+					Command: "mysqld_exporter",
+					Env:     []subprocessmanager.EnvConfig{},
+				},
+			},
+			want: "custom",
+		},
+		{
+			name: "custom name with slashes inside",
+			config: &Config{
+				ReceiverSettings: configmodels.ReceiverSettings{
+					TypeVal: "prometheus_exec",
+					NameVal: "prometheus_exec/custom/name",
+				},
+				ScrapeInterval: 60 * time.Second,
+				Port:           9104,
+				SubprocessConfig: subprocessmanager.SubprocessConfig{
+					Command: "mysqld_exporter",
+					Env:     []subprocessmanager.EnvConfig{},
+				},
+			},
+			want: "custom/name",
+		},
+	}
+
+	for _, test := range customNameTests {
+		t.Run(test.name, func(t *testing.T) {
+			got := extractName(test.config)
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("getCustomName() got = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestGenerateRandomPort(t *testing.T) {
+	t.Run("TestGenerateRandomPort", func(t *testing.T) {
+		testPort := 35000
+		holdPort, _ := net.Listen("tcp", fmt.Sprintf(":%v", testPort))
+		got, err := generateRandomPort()
+		if err != nil {
+			t.Errorf("generateRandomPort() returned an error: %w", err)
+		}
+		if got == testPort {
+			t.Errorf("generateRandomPort() got = %v, wanted something different since this port is in use", got)
+		}
```

Comment:
Honestly wasn't sure how to test this, I know it's highly unlikely but to me it was the only way of testing that it would not pick that port if it's in use... any ideas how I can make this better?